### DISCUSSION TITLE
Combat, spells that cannot be cast, display the information window directly in the Magic Book

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#10304] [General]
In the original game, when the hero cannot cast a spell, for example the Resurrect spell when he has no troop to resurrect or the Dispel Magic spell when there is no active spell, an information window stating this is displayed on the battlefield after the Magic Book has been closed.
This the same in fheroes2.

I suggest that these message windows are displayed directly in the Magic Book (as is the message window telling the hero that he does not have enough magic points to cast a spell), when the player left clicks on the spell and also this will be indicated in the information window displayed when the player right clicks on the spell.

The information bar at the bottom of the screen also displays a message to indicate whether the spell can be cast when the player hovers the mouse cursor over the graphics of different spells.

This will prevent the player from having to reopen the Magic Book if the spell cannot be cast.

In fheroes2:

![Spells that cannot be cast 01](https://user-images.githubusercontent.com/43583994/93310891-a1206080-f805-11ea-9154-090e77210b7d.png)

![Spells that cannot be cast 02](https://user-images.githubusercontent.com/43583994/93310903-a41b5100-f805-11ea-915a-f1a6a67e68f0.png)

**Comment by ihhub** [comment]
Isn't the rest of games in the series have the same behavior? I think making spells unavailable is misleading because a player could think that not enough spell points. For example, a player opens the magic book and sees that a spell A with 10 spell points are greyed out. Then logically they will think that spell B on the second page also greyed out which isn't true. This would be very misleading for new players.

Let's move it to the Discussion section.

**Reply by LeHerosInconnu** [reply]
Hello @ihhub,

???
This proposal does not concern grayed out spells, but rather displaying the message directly in the magic book rather than closing the magic book and displaying the message on the combat area.
I think there is some confusion with this issue: https://github.com/ihhub/fheroes2/discussions/4845.
The proposal presented here can be made separately from the other one.

P.S.: Are you an AI driven bot?
Because I'm wondering, especially with this other case: https://github.com/ihhub/fheroes2/pull/10182. :D

00100000 01000000 01101001 01101000 01101000 01110101 01100010 00111010 00100000 01001001 01100110 00100000 01111001 01101111 01110101 00100000 01100001 01110010 01100101 00100000 01100001 00100000 01100010 01101111 01110100 00101100 00100000 01110000 01101100 01100101 01100001 01110011 01100101 00100000 01110011 01110100 01101111 01110000 00100000 01110100 01101000 01101001 01110011 00101110 00100000 00111010 01000100
